### DISCUSSION TITLE
Update Deck View via Viewport

### DISCUSSION THREAD

**Discussion by dennemark** [visgl/deck.gl#8569] [Q&A]
Hi,
I want to synchronize a 3d viewport with deck.gl viewport. I do not want the camera to be controlled by deck.gl, so I need a way to update the view of deck.gl.
My current approach is:

```
  const deck = new Deck({
        initialViewState: {
          longitude: 13.4,
          latitude: 52.48,
          zoom: 12,
          position: [0,0,10],
          pitch: 0,
          bearing: 0,
          far: 10000
        },
        views: firstPersonView,
       layers: [...]
})

firstPersonView.makeViewport({ 
        width: engine.getRenderWidth(),
        height: engine.getRenderHeight(),
        viewState: new FirstPersonViewport(getViewportFromCamera(camera, engine))
      })
deck.redraw("reason")
```

But this has no effect on the deck canvas. It just stays as it is. But deck.props.onAfterRender definitely logs the redraw.
How could i update the first person view and tell deck to redraw the canvas?

Let me know if you need more information. Thanks!

Edit:
Hm this seems to work, but it is weird to set it this way.
deck.getViewports()[0] = fpViewport

**Comment by Pessimistress** [comment]
https://deck.gl/docs/developer-guide/interactivity#externally-manage-view-state

**Reply by dennemark** [reply]
Hi,
Thanks for your reply!
I am not using react. I checked the code though https://github.com/visgl/deck.gl/blob/1d9b7be1d8bd740d02984f1bc25c10781800b4b3/modules/react/src/deckgl.ts
I cannot find a line in those files where the viewState is being set. I only have my phone today, so i might have overlooked it. But i searched the page and it is not really obvious where the viewState is being set :/

**Reply by Pessimistress** [reply]
The React component is only a thin wrapper of the Deck class. In vanilla JS you just call `deckInstance.setProps({viewState})`.

**Reply by dennemark** [reply]
Thanks! I am going to try! I used version 9 beta5. So i did kot see it in the typings yet. Going to look closer at it!
